src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1774,13 +1782,31 @@ else if (con.getTrustManagerClass() != null) {
             if (logger.isLoggable(Level.FINEST))
                 logger.finest(toString() + " Getting TLS or better SSL context");
 
-            sslContext = SSLContext.getInstance(sslProtocol);
-            sslContextProvider = sslContext.getProvider();
+            if (null != clientCertificate) {
```

Comment:
The code in if/else are almost the same. You just need to initialize `km` to `null` and have `km = SQLServerCertificateUtils.getKeyManagerFromFile(clientCertificate, clientKey,clientKeyPassword);` inside if block.

Reply:
thanks. done